modules/services/utils/src/main/java/au/com/shiftyjelly/pocketcasts/utils/FileUtil.kt

```diff
@@ -0,0 +1,103 @@
+package au.com.shiftyjelly.pocketcasts.utils
+
+import android.content.Context
+import android.content.Intent
+import android.content.pm.PackageManager
+import android.net.Uri
+import androidx.core.content.FileProvider
+import java.io.File
+import java.io.OutputStream
+import okio.IOException
+import okio.buffer
+import okio.sink
+import okio.source
+import timber.log.Timber
+
+object FileUtil {
+    fun deleteFileByPath(path: String) {
+        try {
+            val file = File(path)
+            if (file.exists()) {
+                if (!file.delete()) {
+                    Timber.e("Could not delete file $file")
+                }
+            }
+        } catch (e: Exception) {
+            Timber.e(e, "Could not delete file $path")
+        }
+    }
+
+    fun deleteDirContents(path: String) {
+        try {
+            val dir = File(path)
+            if (!dir.isDirectory) {
+                return
+            }
+
+            dir.listFiles()?.forEach { file ->
+                if (!file.name.equals(".nomedia", ignoreCase = true)) {
+                    file.delete()
+                }
+            }
+        } catch (e: Exception) {
+            Timber.e(e, "Could not delete directory $path contents")
+        }
+    }
+
+    fun readFileTo(file: File, output: OutputStream) {
+        file.source().buffer().use { source ->
+            source.readAll(output.sink())
+        }
+    }
+
+    fun copy(src: File, dst: File) = when {
+        src.isDirectory && dst.isFile -> throw IOException("Can't copy from dir $src to file $dst")
+        src.isFile && dst.isDirectory -> throw IOException("Can't copy from file $src to file $dst")
```

Comment:
I added preemptive checks here to fail fast that throws `IOException`. While the code previously thrown `FileNotFoundException` it doesn't really matter as at all call sites we catch `IOException`.